system-tests/tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/utils/FileTransferSimulationUtils.java

```diff
@@ -0,0 +1,149 @@
+        } catch (IOException e) {
+            throw new EdcException(e);
+        }
+        return group("Contract negotiation")
+                .on(exec(
+                        // Initiate a contract negotiation
+                        http("Contract negotiation")
+                                .post("/api/negotiation")
+                                .body(StringBody(body))
+                                .header(CONTENT_TYPE, "application/json")
+                                .queryParam(CONNECTOR_ADDRESS_PARAM, connectorAddress)
+                                .check(status().is(SC_OK))
+                                .check(bodyString()
+                                        .notNull()
+                                        // UUID is returned to get the contract agreement negotiated between provider and consumer.
+
+                                        .saveAs("contractNegotiationRequestId"))
+                ))
+                // Call ContractNegotiation status endpoint every second, till it gives a CONFIRMED state.
+                .exec(session -> session.set("status", -1))
+                .group("Wait for agreement").on(
+                        doWhileDuring(session -> session.getString("contractAgreementId") == null,
+                                Duration.ofSeconds(30))
+                                .on(exec(http("Get status")
+                                                .get(session -> format("/api/control/negotiation/%s", session.getString("contractNegotiationRequestId")))
+                                                .header(API_KEY_HEADER, apiKey)
+                                                .check(status().is(SC_OK))
+                                                .check(
+                                                        jmesPath("id").is(session -> session.getString("contractNegotiationRequestId")),
+                                                        jmesPath("state").saveAs("status")
+                                                )
+                                                .checkIf(
+                                                        session -> Integer.toString(ContractNegotiationStates.CONFIRMED.code()).equals(session.getString("status"))
+                                                )
+                                                .then(
+                                                        // Obtain contract agreement ID
+                                                        jmesPath("contractAgreement.id").notNull().saveAs("contractAgreementId")
+                                                )
+                                        )
+                                                .pace(Duration.ofSeconds(1))
+                                )
+
+                )
+                .feed(endlesslyWith(() -> Map.of("fileName", FAKER.lorem().characters(20, 40))))
```

Comment:
just curious to know why this .feed is needed with endless iterator with fake filename map.

Reply:
Thank you, it makes sense when pushing up the number of clients, but since this is not in scope of this PR yet, I've removed it - to be re-added in performance test story